server/etcdmain/config.go

```diff
@@ -238,14 +244,44 @@ func (cfg *config) configFromCmdLine() error {
 		cfg.ec.InitialCluster = ""
 	}
 
+	// SetFeatureGatesFromExperimentalFlags validates that cmd line flags for experimental feature and their feature gates are not explicitly set simultaneously,
+	// and passes the values of cmd line flags for experimental feature to the server feature gate.
+	err = embed.SetFeatureGatesFromExperimentalFlags(cfg.ec.ServerFeatureGate, cfg.getBoolFlagVal, embed.ServerFeatureGateFlagName, cfg.cf.flagSet.Lookup(embed.ServerFeatureGateFlagName).Value.String())
+	if err != nil {
+		return err
+	}
+
 	return cfg.validate()
 }
 
+// getBoolFlagVal returns the value of the a given bool flag is explicitly set in the cmd line arguments,
+// and returns nil if it is not explicitly set.
+func (cfg *config) getBoolFlagVal(flagName string) *bool {
+	explicitlySet := false
+	cfg.cf.flagSet.Visit(func(f *flag.Flag) {
+		if f.Name == flagName {
+			explicitlySet = true
+		}
+	})
+	if !explicitlySet {
+		return nil
+	}
+	flagVal, parseErr := strconv.ParseBool(cfg.cf.flagSet.Lookup(flagName).Value.String())
+	if parseErr != nil {
+		panic(parseErr)
+	}
+	return &flagVal
+}
```

Comment:
We can move this into https://github.com/etcd-io/etcd/blob/main/pkg/flags/flag.go as a generic function something like below,

```
func GetBoolFlagVal(fs *flag.FlagSet, flagName string) *bool {
    ......
}
```

Reply:
done.